lib/devise/rails/routes.rb

```diff
@@ -415,7 +415,12 @@ def devise_omniauth_callback(mapping, controllers) #:nodoc:
 ERROR
         end
 
-        path, @scope[:path] = @scope[:path], nil
+        path = @scope[:path]
```

Comment:
I missed #3701 - will take a look, although using frame may be cleaner code for backwards compatability